### PR TITLE
Bluetooth: Mesh: AES-CCM: Fix output MIC with additional data

### PR DESCRIPTION
**Pull request by RehamTarekAhmed**
In case of providing additional authentication data, the bt_mesh_ccm_encrypt function doesn't produce correct MIC values according to RFC3610. This fix was verified using the test vectors that comply with the required MIC sizes.
Signed-off-by: Reham Tarek <reham.tarek@si-vision.com>

### CODE REVIEW

**Review by jhedberg — DISMISSED** [review]
Thanks for this contribution! Could you please include the pull request description in the commit message as well? That way we get the background information into the git commit history. Also, please prefix the commit message with "Bluetooth: Mesh: " - that way it's clear what subsystem this is for. You'll probably need to shorten it however to avoid having a too long line, e.g. "Bluetooth: Mesh: AES-CCM: Fix output MIC with additional data".

I'm also curious why this hasn't been caught with the PTS tests - I think it should be triggerable using virtual label UUIDs, since AFAIK those will cause additional AES-CCM authentication data to be used. Did you try running the respective PTS tests with this to see if they still pass?

**Comment by RehamTarekAhmed** [comment]
Thanks for the response. Unfortunately, I wasn't able to try the PTS tests. I was comparing the AES-CCM MIC output to some test vectors and only became suspicious when they didn't match.

**Comment by RehamTarekAhmed** [comment]
It passed the PTS tests because the virtual address size is 16 octets, MIC only fails for the cases with 14 octet authentication data and below.  Thus the fix may not be required in the cases that mesh uses.

**Comment by jhedberg** [comment]
@RehamTarekAhmed could you still update the commit message body - it's empty currently. The original PR description would be good there, as well as the information that this only concerns AAD values up to 14 bytes of length (which Mesh doesn't currently use).

**Comment by RehamTarekAhmed** [comment]
> @RehamTarekAhmed could you still update the commit message body - it's empty currently. The original PR description would be good there, as well as the information that this only concerns AAD values up to 14 bytes of length (which Mesh doesn't currently use).
It is done now, thanks for your patience.

**Comment by aescolar** [comment]
@RehamTarekAhmed : Please note that the commit message lacks a Signed-Off-by line, without it, it cannot be merged in.

**Comment by aescolar** [comment]
Hi @RehamTarekAhmed , almost there :). The "author" information added by git to the commit needs to match the commit signed off line.
You can just do:
```git commit --amend --author="Reham Tarek <reham.tarek@si-vision.com>"```
(To not have this problem in later commits, you can correct your .gitconfig)

**Comment by RehamTarekAhmed** [comment]
@aescolar Thanks for the heads up

**Review by jhedberg — APPROVED** [review]

**Comment by aescolar** [comment]
@jhedberg : Given that we are in code freeze, even that in a way, this is a bugfix, I guess merging should be help until after the release?

**Comment by jhedberg** [comment]
@aescolar I can't even merge this now even if I wanted to since minimum two approvals are needed. However yes, I'd say this should wait until the release since it doesn't fix any real bug with a current code, rather only a theoretical one that would be introduced if this code was used in outside of Mesh with less than 14 bytes (but greater than 0) of additional data.

**Pull request merged**
